File: backend/java/Vehicle_Configurator/src/main/java/com/example/demo/entities/Segment.java
package com.example.demo.entities;

import jakarta.persistence.*;

@Entity
@Table(name = "segment_master")
public class Segment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "segment_id")
    private int segmentId;

    @Column(name = "segment_name")
    private String segmentName;

    public Segment() {
    }

    public Segment(int segmentId, String segmentName) {
        this.segmentId = segmentId;
        this.segmentName = segmentName;
    }

    public int getSegmentId() {
        return segmentId;
    }

    public void setSegmentId(int segmentId) {
        this.segmentId = segmentId;
    }

    public String getSegmentName() {
        return segmentName;
    }

    public void setSegmentName(String segmentName) {
        this.segmentName = segmentName;
    }
}

File: backend-dotnet/Models/Segment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend_dotnet.Models
{
    [Table("segment_master")]
    public class Segment
    {
        [Key]
        [Column("segment_id")]
        public int segmentId { get; set; }

        [Column("segment_name")]
        public string segmentName { get; set; } = string.Empty;
    }
}
